### DISCUSSION TITLE
Preventing a javascript script from being loaded twice

### DISCUSSION THREAD

**Discussion by nkosi23** [unpoly/unpoly#428] [Q&A] [answered]
Hello,

Thank you for this awesome library!!! Someone should create a cap bearing the text: "Make Server-Side Rendering Great Again" :smile: 

I was wondering, out-the-box is unpoly able to automatically prevent javascript files from being loaded twice?  I am thinking about the common situation in which the server-side template contains the script tags (referencing independent javascript files - I am not talking about inline javascript) needed by a specific page. Sometimes those scripts may be web-components reused across multiple pages (or already loaded by a prior page). So it would be great to be able to ensure that such scripts are not loaded twice.

Thanks

**Comment by triskweline** [comment, marked as answer]
There's no built-in mechanism for this. Maybe this can help: [Unpoly: Loading large libraries on-demand](https://makandracards.com/makandra/52361-unpoly-loading-large-libraries-on-demand)

**Reply by philippejadin** [reply]
This would be a great addition imho. Something like <script up-include > or <script up-include-once> 
I like the idea of adding the required scripts in the view partials I send from the server (laravel blade views & components). Or is it a bad pattern?
